lib/src/Pos/Launcher/Resource.hs

```diff
@@ -296,7 +298,11 @@ allocateNodeContext ancd txpSettings = do
     -- TODO synchronize the NodeContext peers var with whatever system
     -- populates it.
     peersVar <- newTVarIO mempty
-    ncSubscriptionKeepAliveTimer <- newTimer $ 30 * 1000000 -- TODO: use slot duration
+    let slotDuration :: Integer
+        slotDuration = toMicroseconds . bvdSlotDuration $ gdBlockVersionData genesisData
```

Comment:
It's genesis slot duration, it can change over time. You need `getCurrentEpochSlotDuration`, see https://github.com/input-output-hk/cardano-sl/pull/1852#issuecomment-342189339